### DISCUSSION TITLE
@mantine/form - get all dirty form values

### DISCUSSION THREAD

**Discussion by cschreier** [mantinedev/mantine#6889] [Q&A] [answered]
Is it possible to retrieve all dirty fields of a form with a single function?

There is currently the ‘getDirty’ function, but it only returns an empty object.

Currently I have to query each individual field and handle it accordingly.
```
const changedProperties = [];

if (form.isDirty("property1")) {
    changedProperties.push({
        property: "property1",
        value: formValues.property1.trim(),
    });
}
if (form.isDirty("property2")) {
    changedProperties.push({
        property: "property2",
        value: formValues.property2.trim(),
    });
}
```

**Comment by rtivital** [comment]
You can get dirty status of fields with `form.getDirty()`.

**Reply by cschreier** [reply]
Hi @rtivital, 
with `form.getDirty()` I'm getting always an empty object.

**Comment by cschreier** [comment]
@rtivital do you have a solution?

**Reply by rtivital** [reply]
Form dirty status is calculated dynamically for each field, it is not stored in the form.

**Reply by cschreier** [reply]
@rtivital Unfortunately, I can't do much with this answer. Is there a way to retrieve all dirty fields of the form at once, or do I have to check each field individually as described above?

**Reply by rtivital** [reply, marked as answer]
You need to get status of each field individually
